pkg/manager/internal.go

```diff
@@ -437,7 +437,6 @@ func (cm *controllerManager) serveHealthProbes(stop <-chan struct{}) {
 
 func (cm *controllerManager) Start(stop <-chan struct{}) error {
 	// join the passed-in stop channel as an upstream feeding into cm.internalStopper
-	defer close(cm.internalStopper)
```

Comment:
Would this work as an alternative? This should wait for the done channel on all exits from this function without having to explicitly write it at all exits from the function
```
doneCh := make(chan error, 1)
defer func() {
  close(cm.internalStopper)
  <-doneCh
}()
```